src/components/controls/ColorMapControls.jsx

```diff
@@ -0,0 +1,53 @@
+import { Dropdown } from "react-bootstrap";
+
+import { useControlsContext } from "../../context/controls-context";
+
+export default function ColorMapControl(props) {
+  const {
+    state: { allColorMaps, colorMap },
+    dispatch,
+  } = useControlsContext();
+
+  return (
+    <div className="fullWidth">
+      <h4>Color Map</h4>
+      <Dropdown>
+        <Dropdown.Toggle variant="outline-primary" className="fullWidth">
+          <img
+            src={colorMap.src}
+            alt="Selected color map"
+            height="15"
+            width="65%"
+            className="mr-2"
+          />
+          {colorMap.name}
+        </Dropdown.Toggle>
+        <Dropdown.Menu>
+          {allColorMaps.map((color) => {
```

Comment:
If this is getting hard to read, you can do the `map` call inside the main body of the component and render it down here

Reply:
I love creating arrays of JSX elements and then stuffing them in the JSX later. Don't feel like you have to keep all the JSX in the return/render